src/m2n/GatherScatterCommunication.cpp

```diff
@@ -65,101 +66,116 @@ void GatherScatterCommunication::send(precice::span<double const> itemsToSend, i
 {
   PRECICE_TRACE(itemsToSend.size());
 
-  // Gather data
+  // Gather data on secondary ranks
   if (utils::IntraComm::isSecondary()) { // Secondary rank
     if (!itemsToSend.empty()) {
+      PRECICE_DEBUG("Providing {} elements to gather step", itemsToSend.size());
       utils::IntraComm::getCommunication()->send(itemsToSend, 0);
     }
-  } else { // Primary rank or coupling mode
-    PRECICE_ASSERT(utils::IntraComm::getRank() == 0);
-    mesh::Mesh::VertexDistribution &vertexDistribution = _mesh->getVertexDistribution();
-    int                             globalSize         = _mesh->getGlobalNumberOfVertices() * valueDimension;
-    PRECICE_DEBUG("Global Size = {}", globalSize);
-    std::vector<double> globalItemsToSend(globalSize);
-
-    // Primary rank data
-    for (size_t i = 0; i < vertexDistribution[0].size(); i++) {
-      for (int j = 0; j < valueDimension; j++) {
-        globalItemsToSend[vertexDistribution[0][i] * valueDimension + j] += itemsToSend[i * valueDimension + j];
-      }
-    }
+    return;
```

Comment:
Why replacing `else` by `return`?

Reply:
This makes it clear that there is nothing happening after the else.
It makes functions way easier to read.